web/src/main/java/org/cbioportal/web/parameter/StudyViewFilter.java

```diff
@@ -61,6 +62,22 @@ private boolean isEitherValueOrRangePresentInGenomicDataIntervalFilters() {
         return invalidCount == 0;
     }
 
+    @AssertTrue
+    private boolean isEitherValueOrRangePresentInGenericAssayDataIntervalFilters() {
+        long invalidCount = 0;
+
+        if (genericAssayDataFilters != null) {
+            invalidCount = genericAssayDataFilters
+                    .stream()
+                    .flatMap(f -> f.getValues().stream())
+                    .filter(Objects::nonNull)
+                    .filter(v -> v.getValue() != null == (v.getStart() != null || v.getEnd() != null))
```

Comment:
I would think a test to see if value or range is present would be 
```java
v.getValue() != null || (v.getStart() != null || v.getEnd() != null)
```

By checking that the two checks evaluate to the same boolean, you're marking a rows as invalid when it has both a range and a value, or neither a range nor a value. Is that what you want?

Reply:
looks correct to me. If two checks evaluate to the same boolean then it is invalid, which we are evaluating.